src/ranking.jl

```diff
@@ -171,3 +171,174 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(v, value; method=:inc)
+
+Compute the quantile(s)-position in [0, 1] interval of a `value` relative to a collection 
+`v`, e.g. a quantile rank of x means that (x*100)% of the elements in `v` are lesser 
+(strict) or lesser-equal (weak) the given `value`. 
+
+The function gets the counts of elements of `v` that are less than `value` (`count_less`), 
+elements of `v` that are equal to `value` (`count_equal`) and the length of `v` (`n`). 
+It also extracts the highest value of elements below `value` (`last_less`) and the lowest 
+value of elements above `value` (`less_greater`). Using them, different methods and 
+definitions of the `quantilerank` are obtained by changing the `method` keyword argument: 
+
+`:inc` (default) - It calculates a value in the range 0 to 1 inclusive. 
+If `value ∈ v`, it returns `count_less / (n - 1)`, if not, apply interpolation based on 
+def. 7 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK` and `PERCENTRANK.INC`)
+
+`:exc` - It calculates a value in the range 0 to 1 exclusive. 
+If `value ∈ v`, it returns `(count_less + 1) / (n + 1)`,  if not, apply interpolation 
+based on def. 6 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK.EXC`)
+
+`:compete` - Based on the `competerank` function from StatsBase.jl, if `value ∈ v`, 
+it returns `count_less / (n - 1)`, if not, returns `(count_less - 1) / (n - 1)`.
+Also, there is no interpolation.
+(equivalent to MariaDB `PERCENT_RANK`, dplyr `percent_rank`)
+
+`:tied` - Based on the def. in Roscoe, J. T. (1975), it returns 
+`(count_less + count_equal/2) / n` and there is no interpolation. 
+(equivalent to `mean` argument of Scipy `percentileofscore`)
+
+`:strict` - It returns `count_less / n` and there is no interpolation.
+(equivalent to `strict` method of Scipy `percentileofscore`)
+
+`:weak` - It returns `(count_less + count_equal) / n` and there is no interpolation.
+(equivalent to `weak` method of Scipy `percentileofscore`)
+
+!!! note
+    An `ArgumentError` is thrown if `v` contains `NaN` or `missing` values
+    or if `v` is empty or contains only one element.
+
+# References
```

Comment:
Can you mention which reference covers which definition? For example, AFAICT Wikipedia only gives one definition.

Reply:
I didn't think about linking the methods in the "references" section, I did the opposite, I cited the reference in the method text (of all those that there are references).

If you want to change, can you guide me how you are thinking about this reference?

Reply:
That's good for academic references, but Wikipedia isn't cited above, and yet it only deals with a particular method. Saying "covers definition" is a bit misleading.

Reply:
yeah. Now I got you. sorry for my mistake. 

Well, this really stop me. 
I thinked a lot on how could I document the function and my best was this structure. 
I like the ideia of remove the wikipedia references (also for quantile) and keep with the "academic structure", even if it's only for :inc and :exc methods, because this seems the most honestly solution by keeping the structure.

Reply:
OK, let's do that.